x-pack/plugins/security_solution/server/lib/detection_engine/rule_ad_hoc_runner/api/logic/get_invocation_count_from_time_range.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Unit } from '@kbn/datemath';
+import moment from 'moment';
+
+export const getInvocationCountFromTimeRange = ({
```

Comment:
are these functions copied from preview functionality?

Would be nice to make amount of code duplication as small as possible, as it would increase cost of maintenance.
Especially, given a fact, there are no tests for ad-hoc yet